src/main/java/org/ladocuploader/app/cli/PGPEncryptor.java

```diff
@@ -0,0 +1,154 @@
+      InputStream inputStream = PGPUtil.getDecoderStream(fileInputStream);
+      KeyFingerPrintCalculator fpCalculator = new JcaKeyFingerprintCalculator();
+      PGPSecretKeyRingCollection pgpSec = new PGPSecretKeyRingCollection(inputStream, fpCalculator);
+
+      Iterator<PGPSecretKeyRing> keyRingIter = pgpSec.getKeyRings();
+      while (keyRingIter.hasNext()) {
+        PGPSecretKeyRing keyRing = keyRingIter.next();
+
+        Iterator<PGPSecretKey> keyIter = keyRing.getSecretKeys();
+        while (keyIter.hasNext()) {
+          PGPSecretKey key = keyIter.next();
+
+          if (key.isSigningKey()) {
+            return key;
+          }
+        }
+      }
+    } catch (PGPException e) {
+      throw new IllegalArgumentException("Invalid signing key", e);
+    }
+
+    throw new IllegalArgumentException("Invalid signing key");
+  }
+
+  private PGPPrivateKey getPrivateKey(PGPSecretKey secretKey) throws PGPException {
+    return secretKey.extractPrivateKey(
+        new JcePBESecretKeyDecryptorBuilder().build(sigkeyPassword.toCharArray()));
+  }
+
+  private byte[] signAndEncryptPayload(InputStream inputStream, PGPSecretKey secKey, PGPPublicKey pubKey,
+                                      ByteArrayOutputStream outputStream) throws PGPException, IOException {
+    int BUFFER_SIZE = 1 << 16;
+
+    // Encryption
+    PGPEncryptedDataGenerator encGen = new PGPEncryptedDataGenerator(
+        new JcePGPDataEncryptorBuilder(SymmetricKeyAlgorithmTags.AES_256)
+            .setWithIntegrityPacket(true)
+            .setSecureRandom(new SecureRandom()));
+    encGen.addMethod(new JcePublicKeyKeyEncryptionMethodGenerator(pubKey));
+    OutputStream encOut = encGen.open(outputStream, new byte[BUFFER_SIZE]);
+
+    // Compression
+    PGPCompressedDataGenerator cGen = new PGPCompressedDataGenerator(CompressionAlgorithmTags.ZIP);
+    OutputStream cOut = cGen.open(encOut);
+
+    // Signing
+    PGPSignatureGenerator sGen = new PGPSignatureGenerator(new JcaPGPContentSignerBuilder(secKey.getPublicKey().getAlgorithm(), PGPUtil.SHA1));
+    sGen.init(PGPSignature.BINARY_DOCUMENT, getPrivateKey(secKey));
+
+    Iterator<String> it = secKey.getPublicKey().getUserIDs();
+    if (it.hasNext()) {
+      PGPSignatureSubpacketGenerator spGen = new PGPSignatureSubpacketGenerator();
+      spGen.addSignerUserID(false, it.next());
+      sGen.setHashedSubpackets(spGen.generate());
+    }
+
+    sGen.generateOnePassVersion(false).encode(cOut);
+
+    // Literal Data generator and output stream
+    byte[] data = inputStream.readAllBytes();
+    PGPLiteralDataGenerator lGen = new PGPLiteralDataGenerator();
+    OutputStream lOut = lGen.open(cOut, PGPLiteralData.BINARY, PGPLiteralData.CONSOLE, data.length, new Date());
+
+    lOut.write(data);
+    sGen.update(data);
+
+    lOut.close();
+    lGen.close();
+
+    sGen.generate().encode(cOut);
+    cOut.close();
+    cGen.close();
+    encGen.close();
```

Comment:
does any of this write to disk and leave behind anything we should clean up?

Reply:
It shouldn't but let me double check

Reply:
All the streams are closed in this method or from the caller and doesnt write any local files so we should be good